jenkins/bootstrap.py

```diff
@@ -845,9 +845,10 @@ def setup_magic_environment(job, call):
     # Try to set SOURCE_DATE_EPOCH based on the commit date of the tip of the
     # tree.
     # This improves cacheability of stamped binaries.
-    head_commit_date = commit_date('git', 'HEAD', call)
-    if head_commit_date:
-        os.environ[SOURCE_DATE_EPOCH_ENV] = head_commit_date.strip()
+    if os.path.exists('.git'):
```

Comment:
WDYT about `git rev-parse --is-inside-work-tree` instead?

Reply:
Unnecessary given that this is right after setup_root, which chdirs us into the necessary repo.

I'm not a fan of how spammy the logged calls are.

Reply:
we don't have to log the call FWIW, I think this should be fine but we've had some surprising things related to this in the past (.git being a file instead of a dir), which makes me a little wary of this potentially being brittle.

Reply:
probably fine since we should be moving off of bootstrap.py anyhow..

Reply:
/cc @ixdy